internal/k8s/controllers/config_controller.go

```diff
@@ -133,7 +139,12 @@ func (r *ConfigReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctr
 	}
 
 	level.Debug(r.Logger).Log("controller", "ConfigReconciler", "rendered config", dumpConfig(cfg))
+	if r.oldConfig != nil && reflect.DeepEqual(r.oldConfig, cfg) {
+		level.Error(r.Logger).Log("controller", "ConfigReconciler", "event", "configuration did not change, ignoring")
+		return ctrl.Result{}, nil
+	}
 
+	r.oldConfig = cfg
```

Comment:
maybe I'm missing something, but isn't setting this before the Handler problematic in the sense that if the Handler fails with a retry error the config would be ignored in the next reconcile (because of the new DeepEqual conditional)?

Reply:
no the consideration is totally fair. Let me change it so we set the current only when the current is processed successfuly

Reply:
done